src/core/thread/network_data_service.cpp

```diff
@@ -195,20 +195,68 @@ Error Manager::GetNextDnsSrpAnycastInfo(Iterator &aIterator, DnsSrpAnycast::Info
 
 Error Manager::FindPreferredDnsSrpAnycastInfo(DnsSrpAnycast::Info &aInfo) const
 {
+    Error               error = kErrorNotFound;
     Iterator            iterator;
     DnsSrpAnycast::Info info;
-    bool                found = false;
+    DnsSrpAnycast::Info maxSeqNumInfo;
+
+    // Determine the entry with largest seq number in two ways:
+    // `aInfo` will track the largest using serial number arithmetic
+    // comparison, while `maxSeqNumInfo` tracks the largest using
+    // normal compassion.
 
     while (GetNextDnsSrpAnycastInfo(iterator, info) == kErrorNone)
     {
-        if (!found || info.IsSequenceNumberAheadOf(aInfo))
+        if (error == kErrorNotFound)
+        {
+            aInfo         = info;
+            maxSeqNumInfo = info;
+            error         = kErrorNone;
+            continue;
+        }
+
+        if (SerialNumber::IsGreater(info.mSequenceNumber, aInfo.mSequenceNumber))
```

Comment:
I think the current strategy should work well. 

Meanwhile, I think datasets in Network Data are ordered. 
So, could we define the preferred Anycast data by defining the algorithm for finding preferred:
1. Visit seq nums in Network Data in order
2. Find the first seq num, and set it as the `preferred`
3. For each following seq nums in order, if it's larger than `preferred` by serial number arithmetic, update `preferred` to it.

I think the final preferred seq num is also well defined by the algorithm without ambiguity. It's also simpler to implement. 

Thoughts? @abtink @jwhui

Reply:
I actually did consider something very similar this before :)....I agree it may be a bit simpler to implement, but I don't think the netdata TLV/sub-TLVs follow certain order and it is up to leader's implementation how to incorporate the new TLVs. Relying the order may not be good since it may change (based on leader's implementation). I think it may be better to be clear about the preferred SN  (and define it based on seq num set alone).

Reply:
SGTM.